src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -263,4 +366,47 @@ protected virtual void Dispose(bool disposing)
             Dispose(false);
         }
     }
+
+    internal class ThrowHelper
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidateTransformBlock(byte[] inputBuffer, int inputOffset, int inputCount)
+        {
+            if (inputBuffer == null)
+                ThrowArgumentNull(ExceptionArgument.inputBuffer);
+
+            if ((uint)inputCount > inputBuffer.Length)
+                ThrowArgumentOutOfRange(ExceptionArgument.inputCount);
+
+            if (inputOffset < 0)
+                ThrowArgumentOutOfRange(ExceptionArgument.inputOffset);
+
+            if ((inputBuffer.Length - inputCount) < inputOffset)
+                ThrowInvalidOffLen();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ValidateTransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, int inputBlockSize)
+        {
+            ValidateTransformBlock(inputBuffer, inputOffset, inputCount);
+
+            if (inputCount < inputBlockSize)
+                ThrowArgumentOutOfRange(ExceptionArgument.inputCount);
+        }
+
+        public static void ThrowArgumentNull(ExceptionArgument argument) => throw new ArgumentNullException(argument.ToString());
+        public static void ThrowArgumentOutOfRange(ExceptionArgument argument) => throw new ArgumentOutOfRangeException(argument.ToString(), SR.ArgumentOutOfRange_NeedNonNegNum);
+        public static void ThrowInvalidOffLen() => throw new ArgumentException(SR.Argument_InvalidOffLen);
+        public static void ThrowObjectDisposed() => throw new ObjectDisposedException(null, SR.ObjectDisposed_Generic);
+        public static void ThrowCryptographicException() => throw new CryptographicException(SR.Cryptography_SSE_InvalidDataSize);
+        internal static void ThrowBase64FormatException() => throw new FormatException();
```

Comment:
Why are some of these public and some internal?

Reply:
Ups. It's manual created methods vs. VS created methods. Forgot to check this. Will update.